### PR TITLE
Bug 1926054: Update local storage docs about storage class name

### PR DESCRIPTION
**Pull request by gnufied**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1926054

### CODE REVIEW

**Comment by gnufied** [comment]
cc @bobfuru @lisa

**Review by bobfuru — APPROVED** [review]
Made a few minor tweaks to align with OCP docs style guide. Also, removed hard wraps due to newer doc guidelines.

@openshift/team-documentation PTAL for peer review.

File: modules/persistent-storage-local-create-cr.adoc [hunk 1]
```diff
@@ -5,10 +5,7 @@
 [id="local-volume-cr_{context}"]
 = Provisioning the local volumes
 
-Local volumes cannot be created by dynamic provisioning. Instead,
-persistent volumes must be created by the Local Storage Operator. This
-provisioner will look for any devices, both file system and block volumes,
-at the paths specified in defined resource.
+Local volumes cannot be created by dynamic provisioning. Instead, persistent volumes must be created by the Local Storage Operator. This provisioner will look for any devices, both file system and block volumes, at the paths specified in defined resource.
```
Comment (kalexand-rh) [inline modules/persistent-storage-local-create-cr.adoc:8]:
```suggestion
Local volumes cannot be created by dynamic provisioning. Instead, persistent volumes must be created by the Local Storage Operator. This provisioner looks for any file system or block volumes devices at the paths specified in defined resource.
```

File: modules/persistent-storage-local-create-cr.adoc [hunk 2]
```diff
@@ -17,13 +14,12 @@ at the paths specified in defined resource.
 
 .Procedure
 
-. Create the local volume resource. This must define the nodes
-and paths to the local volumes.
+. Create the local volume resource. This must define the nodes and paths to the local volumes.
```
Comment (kalexand-rh) [inline modules/persistent-storage-local-create-cr.adoc:17]:
```suggestion
. Create the local volume resource. This resource must define the nodes and paths to the local volumes.
```

File: modules/persistent-storage-local-create-cr.adoc [hunk 3]
```diff
@@ -46,23 +42,19 @@ spec:
...
-defined, then the Local Storage Operator will attempt to find matching disks
-on all available nodes.
-<3> The volume mode, either `Filesystem` or `Block`, defining the type of the
-local volumes.
-<4> The file system that is created when the local volume is mounted for the
-first time.
-<5> The path containing a list of local storage devices to choose from.
-<6> Replace this value with your actual local disks filepath to the LocalVolume resource, such as `/dev/xvdg`. PVs are created for these local disks when the provisioner is deployed successfully.
+<2> Optional: A node selector containing a list of nodes where the local storage volumes are attached. This example uses the node host names, obtained from `oc get node`. If a value is not defined, then the Local Storage Operator will attempt to find matching disks on all available nodes.
+<3> The name of the storage class to use when creating persistent volumes (PV) objects. The Local Storage Operator automatically creates the storage class if it does not exist. Be sure to use a storage class that uniquely identifies this set of local volumes.
```
Comment (kalexand-rh) [inline modules/persistent-storage-local-create-cr.adoc:53]:
```suggestion
<3> The name of the storage class to use when creating persistent volume objects. The Local Storage Operator automatically creates the storage class if it does not exist. Be sure to use a storage class that uniquely identifies this set of local volumes.
```

File: modules/persistent-storage-local-create-cr.adoc [hunk 4]
```diff
@@ -83,23 +75,19 @@ spec:
...
-on all available nodes.
-<3> The volume mode, either `Filesystem` or `Block`, defining the type of the
-local volumes.
-<4> The path containing a list of local storage devices to choose from.
-<5> Replace this value with your actual local disks filepath to the LocalVolume resource, such as `/dev/xvdg`. PVs are created for these local disks when the provisioner is deployed successfully.
-
-. Create the local volume resource in your {product-title} cluster, specifying
-the file you just created:
+<2> Optional: A node selector containing a list of nodes where the local storage volumes are attached. This example uses the node host names, obtained from `oc get node`. If a value is not defined, then the Local Storage Operator will attempt to find matching disks on all available nodes.
+<3> The name of the storage class to use when creating persistent volumes (PV) objects.
```
Comment (kalexand-rh) [inline modules/persistent-storage-local-create-cr.adoc:85]:
```suggestion
<3> The name of the storage class to use when creating persistent volume objects.
```

File: modules/persistent-storage-local-create-cr.adoc [hunk 5]
```diff
@@ -83,23 +75,19 @@ spec:
...
+<4> The volume mode, either `Filesystem` or `Block`, that defines the type of local volumes.
+<5> The path containing a list of local storage devices to choose from.
+<6> Replace this value with your actual local disks filepath to the `LocalVolume` resource, such as `/dev/xvdg`. PVs are created for these local disks when the provisioner is deployed successfully.
+
+. Create the local volume resource in your {product-title} cluster, specifying the file you just created:
```
Comment (kalexand-rh) [inline modules/persistent-storage-local-create-cr.adoc:90]:
```suggestion
. Create the local volume resource in your {product-title} cluster. Specify the file that you just created:
```

**Review by kalexand-rh — COMMENTED** [review]
I have a few suggestions, but this is looking good.

**Comment by mburke5678** [comment]
@gnufied @bobfuru 
I have a PR that affects the _Provisioning the local volumes_ file. For 4.6, my PR will likely add the lines that you added here. Should I remove these additions after cherry-picking to 4.6?
https://github.com/openshift/openshift-docs/pull/29315/files#diff-02193858ec9b2a747d451748889f4f87a2768126247cd0515897031e56a38b66R44
https://github.com/openshift/openshift-docs/pull/29315/files#diff-02193858ec9b2a747d451748889f4f87a2768126247cd0515897031e56a38b66R77

**Comment by gnufied** [comment]
which PR you are talking about? can you open a PR which includes both of these changes, so as we can see it in right context?

**Comment by mburke5678** [comment]
@gnufied Sorry. Here is the PR: 
https://github.com/openshift/openshift-docs/pull/28680/files#diff-02193858ec9b2a747d451748889f4f87a2768126247cd0515897031e56a38b66

This PR ^^ is based on master and has the `storageClassName: "local-sc" <3>` you added here.

**Comment by bobfuru** [comment]
@mburke5678 IIUC, there shouldn't be any need to back out the changes. With this PR merge, the changes weren't backported prior to 4.7 because the issue wasn't a regression. But having the notes in 4.6 shouldn't be a problem. But I'll let @gnufied confirm.

**Pull request merged**